src/compiler/checker.ts

```diff
@@ -26522,6 +26523,31 @@ namespace ts {
             let prop: Symbol | undefined;
             if (isPrivateIdentifier(right)) {
                 const lexicallyScopedSymbol = lookupSymbolForPrivateIdentifierDeclaration(right.escapedText, right);
+
+                if (lexicallyScopedSymbol && (compilerOptions.target === ScriptTarget.ESNext && !useDefineForClassFields)) {
+                    const lexicalValueDecl = lexicallyScopedSymbol.valueDeclaration;
+                    const lexicalClass = getContainingClass(lexicalValueDecl);

```

Comment:
```suggestion
                if (lexicallyScopedSymbol?.valueDeclaration && (compilerOptions.target === ScriptTarget.ESNext && !useDefineForClassFields)) {
                    const lexicalClass = getContainingClass(lexicallyScopedSymbol.valueDeclaration);
```

1. valueDeclaration now forces you to check for undefined
2. inline lexicalValueDecl since it's only used once and its initialiser isn't that complicated.